### ISSUE TITLE
TEST: /api/edit ROUTE

### ISSUE THREAD

**Issue by zeim839** [ufosc/Club_Website_2#158]
**TODO:** 
1. Create test/edit.spec.js file.
2. Test /api/blog route

For reference, see the testing scripts for [/api/users](https://github.com/ufosc/Club_Website_2/blob/main/test/users.spec.js). If you're interested on working on this issue, leave a comment and I'll respond with the requirements & specifications.

**Comment by Thinura** [comment]
Hi @zeim839 I would like to contribute to this issue. Could you share more details regarding requirements and specifications?

**Comment by zeim839** [comment]
@Thinura 

Thanks for volunteering! Keep in mind that this is a time consuming issue, so feel free to complete as many or as few specs as you wish. 

The testing file should be created in `test/blog.spec.js`. For testing, we use [MochaJS](https://mochajs.org/) and [ChaiJS](https://www.chaijs.com/). Refer to test/users.spec.js for an example.

Each of the following specifications (demarcated with a '-' prefix) should constitute a separate Mocha test, i.e: 
```js
it('Should ...') { /* ... */ }
````

Refer to [README](https://github.com/ufosc/Club_Website_2#readme) and [CONTRIBUTING](https://github.com/ufosc/Club_Website_2/blob/main/CONTRIBUTING.md) for instructions on setting up the project and contributing.

### Testing specifications:

Route: GET /:id
- Should return the 'article not found' page for invalid IDs
An invalid is any non-hex string of less than 24 characters.
- Should return the 'article not found' page for IDs that dont exist in DB.
- Should return the 'article not found' page if the ID exists, the article status is not 'published', and the user is not authenticated.
- Should return the correct blog post when ID exists and status is 'published'.

Route: DELETE /:id
- Should return an error if client is not authenticated.
- Should return an error if an ID is invalid.
- Should return an error if an ID does not exist.
- Should delete blog post from DB when user is authenticated and ID exists.
  and respond with the deleted blog data.

Route: PUT /:id
- Should return an error if client is not authenticated.
- Should return an error if an ID is invalid.
- Should return an error if an ID does not exist.
- Should return an error if client attempts to modify ID.
- Should respond with status 200 and the new edited blog data when user is authenticated and ID exists.
- Should accept any set of model keys (see model/blog.js).
  ex: { title: "myTitle", content: "hello world" }
  Basically allow the user to submit any of the model fields.

Route: POST /
- Should return an error if client is not authenticated.
- Should return an error if required fields are not found (title and status)
- Should return an error if '_id' or 'id' fields are set.
- Should create a new blog post.

Route: GET /
- Should return all blog posts with 'published' status if client is not authenticated.
- Should return all blog posts if client is authenticated.
